src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -0,0 +1,189 @@
+import cherrypy
+from cherrypy._cpserver import Server
+import json
+import logging
+import tempfile
+from typing import Dict, List, TYPE_CHECKING, cast, Collection
+
+from orchestrator import OrchestratorError
+from mgr_module import ServiceInfoT
+from mgr_util import verify_tls_files, build_url
+from cephadm.services.ingress import IngressSpec
+from cephadm.ssl_cert_utils import SSLCerts
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+
+
+def cherrypy_filter(record: logging.LogRecord) -> int:
+    blocked = [
+        'TLSV1_ALERT_DECRYPT_ERROR'
+    ]
+    msg = record.getMessage()
+    return not any([m for m in blocked if m in msg])
+
+
+logging.getLogger('cherrypy.error').addFilter(cherrypy_filter)
+cherrypy.log.access_log.propagate = False
+
+
+class ServiceDiscovery():
+    def __init__(self, mgr: "CephadmOrchestrator") -> None:
+        self.mgr = mgr
+        self.ssl_certs = SSLCerts()
+        self.server_port = self.mgr.service_discovery_port
+        self.server_addr = self.mgr.get_mgr_ip()
+
+    def configure_routes(self) -> None:
+
+        root_server = Root(self.mgr,
+                           self.server_port,
+                           self.server_addr,
+                           self.cert_file.name,
+                           self.key_file.name)
+
+        # configure routes
+        d = cherrypy.dispatch.RoutesDispatcher()
+        d.connect(name='index', route='/', controller=root_server.index)
+        d.connect(name='sd-config', route='/prometheus/sd-config',
+                  controller=root_server.get_sd_config)
+        d.connect(name='rules', route='/prometheus/rules',
+                  controller=root_server.get_prometheus_rules)
+        cherrypy.tree.mount(None, '/sd', config={'/': {'request.dispatch': d}})
```

Comment:
This routing doesn't seem to work for the 'help' page
https://ip:port/ -> 404
https://ip:port/sd -> 404
https://ip:port/sd/ -> 200 and you get the help

just hitting '/' should render the help right?

Reply:
fixed in the last push